tests/dialects/test_bigquery.py

```diff
@@ -262,7 +262,7 @@ def test_bigquery(self):
                 "duckdb": "CAST(a AS BIGINT)",
                 "presto": "CAST(a AS BIGINT)",
                 "hive": "CAST(a AS BIGINT)",
-                "spark": "CAST(a AS LONG)",
```

Comment:
this is great can you also add the spark2 test

Reply:
In looking through the rest of the `test_bigquery.py` all the tests are based on `spark` rather than spark2. Any updates for spark2 should probably be applied in a separate pull request. Would you agree?

Reply:
If you move the changes to the Spark2 dialect, then maybe testing "spark" suffices, since it inherits from Spark2